src/utils/file.cpp

```diff
@@ -165,6 +166,40 @@ int fd_streambuf::underflow() {
 // }}}
 
 namespace file_util {
+  namespace {
```

Comment:
It gives the helper functions internal linkage. Not only does it allow the compiler to be more aggressive when optimizing (mostly in terms of what it will inline), but it also can avoid some weird errors when building with link-time optimization (which are unlikely to happen here, but in general it's helpful).